indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -828,34 +910,49 @@ public TaskStatusPlus apply(TaskRunnerWorkItem workItem)
     }
   }
 
-  private Collection<? extends TaskRunnerWorkItem> securedTaskRunnerWorkItem(
-      Collection<? extends TaskRunnerWorkItem> collectionToFilter,
+  private List<TaskStatusPlus> securedTaskStatusPlus(
+      List<TaskStatusPlus> collectionToFilter,
+      Map<String, String> taskInfoCache,
+      @Nullable String dataSource,
+      @Nullable String type,
       HttpServletRequest req
   )
   {
-    Function<TaskRunnerWorkItem, Iterable<ResourceAction>> raGenerator = taskRunnerWorkItem -> {
-      final String taskId = taskRunnerWorkItem.getTaskId();
-      final Optional<Task> optionalTask = taskStorageQueryAdapter.getTask(taskId);
-      if (!optionalTask.isPresent()) {
+    Function<TaskStatusPlus, Iterable<ResourceAction>> raGenerator = taskStatusPlus -> {
+      final String taskId = taskStatusPlus.getId();
+      String taskDatasource = taskInfoCache.get(taskId);
```

Comment:
I think we can remove the `taskInfoCache`, since it's purely local to `getTasks` now and it's only read from here (we can get the datasource from `taskStatusPlus` as well)

Reply:
right, it can be removed.